pkg/component/controller/apiendpointreconciler_test.go

```diff
@@ -172,7 +172,7 @@ func TestReconcilerWithNeedForUpdate(t *testing.T) {
 	assert.Equal(t, "bar", e.ObjectMeta.Annotations["foo"])
 }
 
-func verifyEndpointAddresses(t *testing.T, expectedAddresses []string, fakeFactory testutil.FakeClientFactory) *corev1.Endpoints {
+func verifyEndpointAddresses(t *testing.T, expectedAddresses []string, fakeFactory *testutil.FakeClientFactory) *corev1.Endpoints {
```

Comment:
This fakeclientfactory business is somehow hard to follow.

Maybe there should be some kind of `ClientGetter` interface and functions like this one would take `clientFactory ClientGetter` instead of `fakeFactory *testutil.FakeClientFactory`).

Reply:
It's `clientFactory kubeutil.ClientFactoryInterface` when it is ultimately used.

Reply:
Accept interfaces, return concrete types :)